Sources/NIO/ConvenienceOptionSupport.swift

```diff
@@ -0,0 +1,178 @@
+    /// - See: channelOption
+    /// - Parameter options: Set of shorthand options to apply.
+    /// - Returns: The updated bootstrap (`self` being mutated)
+    public func options(_ options: NIOTCPShorthandOptions) -> NIOClientTCPBootstrap {
+        var optionsRemaining = options
+        // First give the underlying a chance to consume options.
+        let withUnderlyingOverrides =
+            NIOClientTCPBootstrap(self, withUpdated: underlyingBootstrap._applyOptions(&optionsRemaining))
+        // Default apply any remaining options.
+        return optionsRemaining.applyFallbackMapping(withUnderlyingOverrides)
+    }
+}
+
+// MARK: Utility
+/// Has an option been set?
+/// Option has a value of generic type T.
+public enum NIOOptionValue<T> {
+    /// The option was not set.
+    case notSet
+    /// The option was set with a value of type T.
+    case set(T)
+}
+
+public extension NIOOptionValue where T == () {
+    /// Convenience method working with bool options as bool values for set.
+    var isSet: Bool {
+        get {
+            switch self {
+            case .notSet:
+                return false
+            case .set(()):
+                return true
+            }
+        }
+    }
+}
+
+private extension NIOOptionValue where T == () {
+    init(flag: Bool) {
+        if flag {
+            self = .set(())
+        } else {
+            self = .notSet
+        }
+    }
+}
+
+// MARK: TCP - data
+/// A TCP channel option which can be applied to a bootstrap using shorthand notation.
+public struct NIOTCPShorthandOption: Hashable {
+    fileprivate var data: ShorthandOption
+    
+    private init(_ data: ShorthandOption) {
+        self.data = data
+    }
+    
+    fileprivate enum ShorthandOption: Hashable {
+        case reuseAddr
+        case disableAutoRead
+        case allowRemoteHalfClosure
+    }
+}
+
+/// Approved shorthand options.
+extension NIOTCPShorthandOption {
+    /// Allow immediately reusing a local address.
+    public static let allowImmediateLocalEndpointAddressReuse = NIOTCPShorthandOption(.reuseAddr)
+    
+    /// The user will manually call `Channel.read` once all the data is read from the transport.
+    public static let disableAutoRead = NIOTCPShorthandOption(.disableAutoRead)
+    
+    /// Allows users to configure whether the `Channel` will close itself when its remote
+    /// peer shuts down its send stream, or whether it will remain open. If set to `false` (the default), the `Channel`
+    /// will be closed automatically if the remote peer shuts down its send stream. If set to true, the `Channel` will
+    /// not be closed: instead, a `ChannelEvent.inboundClosed` user event will be sent on the `ChannelPipeline`,
+    /// and no more data will be received.
+    public static let allowRemoteHalfClosure =
+        NIOTCPShorthandOption(.allowRemoteHalfClosure)
+}
+
+/// A set of `NIOTCPShorthandOption`s
+public struct NIOTCPShorthandOptions : ExpressibleByArrayLiteral, Hashable {
```

Comment:
No-one mentioned pushing NIOTCPShorthandOptions into ChannelOptions - I'll assume that was an oversight as it feels like it should logically live there with everything else.